### PR TITLE
Issue #152 - KDBX Version 4

### PR DESCRIPTION
**Pull request by subdavis**
This PR is likely not ready.  There are a few kinks to work out, including some JSON end of string error.

However, version 4 (all combinations of ciphers) are working now.  You can take a look at the code and see what's what.

### CODE REVIEW

**Comment by subdavis** [comment]
Ready for review.

File: popups/controllers/masterPasswordController.js [hunk 1]
```diff
@@ -156,14 +159,21 @@ function MasterPasswordController($scope, $routeParams, $location, keepass, unlo
...
+        passwordKey: undefined,
         keyFileName: $scope.selectedKeyFile ? $scope.selectedKeyFile.name : "",
         rememberPeriod: $scope.rememberPeriod
-      });
+      }
+      if (typeof passwordKey === "object" && $scope.rememberPassword){
+        databaseUsage['passwordKey'] = passwordKey;
+        settings.saveCurrentDatabaseUsage(databaseUsage);
+      } else if ($scope.rememberPassword) {
+        databseUsage['passwordKey'] = Base64.encode(passwordKey);
```
Comment (DukeyToo) [inline popups/controllers/masterPasswordController.js]: Typo on `databseUsage`

**Comment by subdavis** [comment]
Hold off on reviews just yet.  I've rethought a couple things over the weekend and am working on tidying some things up.

File: popups/controllers/masterPasswordController.js (hunk 1 above)
Comment (subdavis) [inline popups/controllers/masterPasswordController.js]: fixed.

**Comment by subdavis** [comment]
Alright, let's see how this looks.  I tried to make the changes as uninvasive as possible.

**Comment by subdavis** [comment]
Bump.   I'll have a lot of time to work on this next week, then probably not so much.

**Comment by subdavis** [comment]
Due to lack of activity, I've forked the project over to https://github.com/suBDavis/CKPX

It is available for install via https://subdavis.com/CKPX/

**Comment by subdavis** [comment]
This work is complete.

**Pull request closed without merging**